modules/pam_unix/unix_chkpwd.c

```diff
@@ -103,6 +104,13 @@ int main(int argc, char *argv[])
 	 */
 	setup_signals();
 
+	/*
+	 * Use landlock to reduce the file scope of unix_chkpwd to /etc/shadow
+	 */
+
+	if (_restrict_to_file_read("/etc/shadow") > 0)
```

Comment:
Is it possible to open `/etc/shadow` before restricting the current thread and use this file descriptor after? This will simplify the whole thing by only restricting access to all files (opened FD are not taken into account).